### PR TITLE
Avoid hardcoding the location of build compiler in `compile-triton.sh`

### PR DESCRIPTION
**Pull request by whitneywhtsang**
Addressed post review comment: https://github.com/intel/intel-xpu-backend-for-triton/pull/421#pullrequestreview-1855082914

### CODE REVIEW

File: scripts/compile-triton.sh [hunk 1]
```diff
@@ -81,8 +81,12 @@ fi
 ############################################################################
 ## Configure and build the llvm project.
 
-C_COMPILER=/usr/bin/gcc
-CXX_COMPILER=/usr/bin/g++
+if [ ! -d "$C_COMPILER" ] || [ ! -d "$CXX_COMPILER" ]; then
```
Comment (pbchekin) [inline scripts/compile-triton.sh:84]: Wrong test `-d` for the variables. Also the logic is not clear: why if one is not set, then set both?
```
if [ -z "$C_COMPILER" ]; then
  C_COMPILER="gcc"
fi
if [ -z "$CXX_COMPILER" ]; then
  CXX_COMPILER="g++"
fi
```

File: scripts/compile-triton.sh [hunk 2]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env bash
 
 set +o xtrace
-if [ ! -d "$BASE" ]; then
+if [ ! -z "$BASE" ]; then
```
Comment (pbchekin) [inline scripts/compile-triton.sh:4]: Bash `-z` is a test for empty string.
```suggestion
if [ -z "$BASE" ]; then
```

File: scripts/compile-triton.sh [hunk 3]
```diff
@@ -81,8 +81,14 @@ fi
 ############################################################################
 ## Configure and build the llvm project.
 
-C_COMPILER=/usr/bin/gcc
-CXX_COMPILER=/usr/bin/g++
+if [ ! -z "$C_COMPILER" ]; then
```
Comment (pbchekin) [inline scripts/compile-triton.sh:84]:
```suggestion
if [ -z "$C_COMPILER" ]; then
```

File: scripts/compile-triton.sh [hunk 4]
```diff
@@ -81,8 +81,14 @@ fi
...
+  C_COMPILER=`which gcc`
+  echo "**** C_COMPILER is set to $C_COMPILER ****"
+fi
+if [ ! -z "$CXX_COMPILER" ]; then
```
Comment (pbchekin) [inline scripts/compile-triton.sh:88]:
```suggestion
if [ -z "$CXX_COMPILER" ]; then
```

**Review by pbchekin — APPROVED** [review]

File: scripts/compile-triton.sh (hunk 2 above)
Comment (whitneywhtsang) [reply scripts/compile-triton.sh]: opps

**Review by etiotto — APPROVED** [review]

**Pull request merged**